### PR TITLE
[App] Improve the autoscaler UI

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
cc @borda @hhsecond

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/lightning_app/components/auto_scaler.py [hunk 1]
```diff
@@ -332,6 +341,70 @@ def send_request_to_update_servers(self, servers: List[str]):
...
+            if v["type"] == "string":
+                out[k] = "data string"
+            elif v["type"] == "number":
+                out[k] = 0.0
+            elif v["type"] == "integer":
+                out[k] = 0
+            elif v["type"] == "boolean":
+                out[k] = False
+            else:
+                raise TypeError("Unsupported type")
```
Comment (Borda) [inline src/lightning_app/components/auto_scaler.py:361]:
```suggestion
        lut = {"string": "data string", "number": 0.0, "integer": 0, "boolean": False}
        for k, v in datatype_props.items():
            if v["type"] not in lut:
                raise TypeError("Unsupported type")
            out[k] = lut[v["type"]]
```

File: src/lightning_app/components/auto_scaler.py [hunk 2]
```diff
@@ -332,6 +341,70 @@ def send_request_to_update_servers(self, servers: List[str]):
...
+        return out
+
+    def get_code_sample(self, url: str) -> Optional[str]:
+        input_type: Any = self._input_type
+        output_type: Any = self._output_type
+
+        if not (hasattr(input_type, "request_code_sample") and hasattr(output_type, "response_code_sample")):
```
[the remark this answers is not in the document]
Comment (akihironitta) [reply src/lightning_app/components/auto_scaler.py:368]: The suggestion here won't do the same thing

**Review by Borda — APPROVED** [review]

File: src/lightning_app/components/serve/__init__.py [hunk 3]
```diff
@@ -1,5 +1,4 @@
-from lightning_app.components.serve.gradio import ServeGradio
-from lightning_app.components.serve.python_server import Category, Image, Number, PythonServer, Text
+from lightning_app.components.serve.python_server import Category, Image, Number, Text, PythonServer, AutoScaler
```
Comment (hhsecond) [inline src/lightning_app/components/serve/__init__.py:1]:
```suggestion
from lightning_app.components.serve.python_server import Category, Image, Number, Text, PythonServer, AutoScaler
from lightning_app.components.serve.gradio import ServeGradio
```

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/lightning_app/components/serve/auto_scaler.py [hunk 4]
```diff
@@ -574,5 +641,8 @@ def autoscale(self) -> None:
         self._last_autoscale = time.time()
 
     def configure_layout(self):
-        tabs = [{"name": "Swagger", "content": self.load_balancer.url}]
+        tabs = [
+            {"name": "Endpoint Info", "content": f"{self.load_balancer}/endpoint-info"},
+            {"name": "Swagger", "content": self.load_balancer.url},
```
Comment (ethanwharris) [inline src/lightning_app/components/serve/auto_scaler.py:646]: Not sure the swagger UI is needed, API access UI should cover it

**Review by ethanwharris — APPROVED** [review]
LGTM, not sure the swagger UI is needed

File: src/lightning_app/components/serve/auto_scaler.py (hunk 4 above)
Comment (hhsecond) [reply src/lightning_app/components/serve/auto_scaler.py:646]: Load balancer also creates a few other endpoints for app developers to see how many concurrent requests are currently active etc. Swagger gives info about those as well

**Review by hhsecond — APPROVED** [review]

**Pull request merged**